mailer.js

```diff
@@ -26,6 +27,31 @@ module.exports = function (log) {
     return 'href="' + url + '" style="color: #0095dd; text-decoration: none; font-family: sans-serif;"'
   }
 
+  function constructLocationString (location) {
+    // construct the location string from the location object
+    var locationString = ''
+    if (location) {
+      if (location.city) {
+        locationString += location.city + ', '
+      }
+      locationString += location.country + ' ' + '(estimated)'
```

Comment:
> Using a conditional sounds OK to me, I assume the same syntax as used in content-server views will also work here.

From an l10n perspective, both the current and proposed schemes are verboten. Both make assumptions about the how the city, country, and `estimated` strings should be combined for all locales.

I propose we use two distinct strings and allow localizers to format the strings as they best see fit:

```js
if (! location) {
  return ''
}

if (location.city) {
  return t('%(city)s, %(country}s (estimated)', location);
} else {
  return t('%(country}s (estimated)', location);
}
```

`t` in this case would need to do the string interpolation.

I believe similar logic can be used within the templates - I have not tried this, so I have no idea:

```html
{{#location}}
  <span class="location">
  {{#location.city}}
     {{t "Location: %(city)s, %(country)s (estimated)", location}}
  {{/location.city}}
  {{^location.city}}
     {{t "Location: %(country)s (estimated)", location}}
  {{/location.city}}
  </span>
{{/location}}
```

It may be that we have to update the template's `t` helper function to do interpolation. It could also be that I don't really know mustache all that well.

Do not follow the [format used in the suspicious_location.html](https://github.com/mozilla/fxa-auth-mailer/blob/master/templates/suspicious_location.html#L39) template. The separate strings in `{{t "Location: }} {{ location }}` were used because those emails were originally written by the legal team and translated by a special legal translating service w/o our input on best practices for l10n. We haven't gone back and updated the templates (see https://github.com/mozilla/fxa-auth-mailer/issues/183).